graph-deploy.js

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env node
+
+let app = require('commander')
+let request = require('request')
+let args = require('./src/cli/args')
+const url = require('url')
+
+args.addBuildCommand()
+
+app
+    .option(
+        '-n, --subgraph-name <NAME>',
+        'subgraph name'
+    )
+    .option(
+        '--node <URL>[:PORT]',
+        'graph node url'
+    )
+
+app.parse(process.argv)
+
+if (!app.node || !app.subgraphName) {
+    app.help()
+}
+
+let compiler = args.compilerFromArgs()
+
+let requestUrl = new URL(app.node);
+if (!requestUrl.port) {
+    requestUrl.port = "8020"
+}
+
+compiler.compile()
+    .then(
+        function (ipfsHash) {
+            if (ipfsHash === undefined) {
+                console.log("compilation failed, not deploying")
+                return
+            }
+            requestUrl.pathname += `subgraph_deploy/${app.name}/${ipfsHash}`
```

Comment:
I'd use `path.join('subgraph_deploy', app.subgraphName, ipfsHash)` here.

BTW, it's `app.subgraphName`, not `app.name`.